code/modules/projectiles/guns/launcher/crossbow.dm

```diff
@@ -131,125 +126,122 @@
-	cell.use(500)
+	to_chat(user, SPAN_NOTICE("[chambered] sparks and crackles as it gives off a red-hot glow."))
+	QDEL_NULL(chambered.BB)
+	chambered.BB = new superheat_type(chambered)
+	cell.use(superheat_cost)
 
-/obj/item/gun/launcher/crossbow/update_icon()
+/obj/item/gun/projectile/crossbow/update_icon()
 	if(tension > 1)
 		icon_state = "crossbow-drawn"
-	else if(bolt)
+	else if(chambered)
 		icon_state = "crossbow-nocked"
 	else
 		icon_state = "crossbow"
 
 /*////////////////////////////
 //	Rapid Crossbow Device	//
 */////////////////////////////
-/obj/item/arrow/RCD
-	name = "flashforged bolt"
-	desc = "The ultimate ghetto 'deconstruction' implement."
-	throwforce = 6
 
-/obj/item/gun/launcher/crossbow/RCD
+/obj/item/gun/projectile/crossbow/RCD
 	name = "rapid crossbow device"
 	desc = "A hacked together RCD turns an innocent construction tool into the penultimate 'deconstruction' tool. Flashforges projectiles using matter units when the string is drawn back."
 	icon = 'icons/obj/guns/launcher/rxb.dmi'
 	icon_state = "rxb"
 	slot_flags = null
-	draw_time = 5
+	draw_time = 7.5
+	superheat_cost = 150 //guild design, more efficient or something
 	var/stored_matter = 0
 	var/max_stored_matter = 60
 	var/boltcost = 5
+	var/obj/item/ammo_casing/flashforge_type = /obj/item/ammo_casing/rod_bolt/rcd
+	superheat_type = /obj/item/projectile/bullet/rod_bolt/rcd/superhot
 
-/obj/item/gun/launcher/crossbow/RCD/proc/genBolt(var/mob/user)
-	if(stored_matter >= boltcost && !bolt)
-		bolt = new/obj/item/arrow/RCD(src)
+/obj/item/gun/projectile/crossbow/RCD/proc/genBolt(var/mob/user)
+	if(stored_matter >= boltcost && !chambered)
+		chambered = new flashforge_type(src)
 		stored_matter -= boltcost
 		to_chat(user, "<span class='notice'>The RXD flashforges a new bolt!</span>")
 		update_icon()
 	else
 		to_chat(user, "<span class='warning'>The \'Low Ammo\' light on the device blinks yellow.</span>")
 		flick("[icon_state]-empty", src)
 
-/obj/item/gun/launcher/crossbow/RCD/attack_self(mob/living/user as mob)
+/obj/item/gun/projectile/crossbow/RCD/attack_self(mob/living/user as mob)
 	if(tension)
 		user.visible_message("[user] relaxes the tension on [src]'s string.","You relax the tension on [src]'s string.")
+		if(chambered)
+			if((stored_matter + 5) > max_stored_matter)
+				to_chat(user, "<span class='notice'>Unable to reclaim flashforged bolt. The RXD can't hold that many additional matter-units.</span>")
+				new /obj/item/arrow/rcd(get_turf(src))
+				return
+			to_chat(user, "<span class='notice'>Reclaimed flashforged bolt.</span>")
+			QDEL_NULL(chambered)
+			stored_matter += 5
 		tension = 0
 		update_icon()
 	else
 		genBolt(user)
 		draw(user)
 
-/obj/item/gun/launcher/crossbow/RCD/attackby(obj/item/W as obj, mob/user as mob)
+/obj/item/gun/projectile/crossbow/RCD/attackby(obj/item/W as obj, mob/user as mob)
 	var/obj/item/stack/material/M = W
 	if(istype(M) && M.material.name == MATERIAL_COMPRESSED_MATTER)
```

Comment:
We truly, **really** need a TG-style RCD that takes any stack of metal or glass and turns it into its own ammo rather than having it _specifically_ be compressed matter tbh....